gitlab/client.py

```diff
@@ -721,6 +722,9 @@ def http_request(
         send_data = self._backend.prepare_send_data(files, post_data, raw)
         opts["headers"]["Content-type"] = send_data.content_type
 
+        if extra_headers is not None:
+            opts["headers"].update(extra_headers)
```

Comment:
When adding the docs can we also mention this can override existing headers set on the instance level?